### PR TITLE
github: Add deployment test

### PR DESCRIPTION
**Pull request by stgraber**
Signed-off-by: Stéphane Graber <stgraber@ubuntu.com>

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+    - name: Build charm
+      run: |
+        set -eux
+        sudo -g lxd charmcraft build
+
+    - name: Bootstrap Juju
+      run: |
+        set -eux
+        sudo -g lxd lxc network set lxdbr0 ipv6.address=none
```
Comment (simondeziel) [inline .github/workflows/test.yml:36]: Despite some documentation still suggesting it, I'm not sure it's needed anymore.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      run: |
+        set -eux
+        juju deploy ./lxd_ubuntu-20.04-amd64.charm --num-units 8 --config sysctl-tuning=false --config kernel-hardening=false
+        juju wait
+        juju status
+
+    - name: Check for errors
+      run: |
+        set -eux
+        if [ -n "$(juju status --format=oneline | grep workload:error)" ]; then
```
Comment (simondeziel) [inline .github/workflows/test.yml:49]: To please shellcheck:

```
if juju status --format=oneline | grep -qF workload:error; then
```

File: .github/workflows/test.yml (hunk 1 above)
Comment (stgraber) [reply .github/workflows/test.yml:36]: It is, unfortunately.

**Comment by stgraber** [comment]
Updated

**Pull request merged**